Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -455,16 +451,14 @@ public typealias StartPurchaseBlock = (@escaping PurchaseCompletedBlock) -> Void
         }
 
         if self.systemInfo.dangerousSettings.autoSyncPurchases {
-            storeKit1Wrapper?.delegate = purchasesOrchestrator
+            self.paymentQueueWrapper.storeKit1Wrapper?.delegate = purchasesOrchestrator
         } else {
             Logger.warn(Strings.configure.autoSyncPurchasesDisabled)
         }
 
         /// If SK1 is not enabled, `PaymentQueueWrapper` needs to handle transactions
         /// for promotional offers to work.
-        if !self.isStoreKit1Configured {
-            self.paymentQueueWrapper.delegate = purchasesOrchestrator
-        }
+        self.paymentQueueWrapper.right?.delegate = purchasesOrchestrator
```

Comment:
Hmm I wonder if we could add another extension to `EitherPaymentQueueWrapper` to access the `PaymentQueueWrapper`. Something like `.paymentQueueWrapper` instead of `.right` to make clearer what it is. So this would become: `self.paymentQueueWrapper.paymentQueueWrapper?.delegate = purchasesOrchestrator`

Reply:
Yeah that's a good suggestion, let me do that.